bundles/org.openhab.binding.chatgpt/src/main/java/org/openhab/binding/chatgpt/internal/ChatGPTHandler.java

```diff
@@ -0,0 +1,200 @@
+        if (response != null) {
+            ChatResponse chatResponse = gson.fromJson(response, ChatResponse.class);
+            if (chatResponse != null) {
+                String msg = chatResponse.getChoices().get(0).getMessage().getContent();
+                updateState(channelUID, new StringType(msg));
+            } else {
+                logger.warn("Didn't receive any response from ChatGPT - this is unexpected.");
+            }
+        }
+    }
+
+    private @Nullable String sendPrompt(ChannelUID channelUID, String prompt) {
+        Channel channel = getThing().getChannel(channelUID);
+        if (channel == null) {
+            logger.error("Channel with UID '{}' cannot be found on Thing '{}'.", channelUID, getThing().getUID());
+            return null;
+        }
+        ChatGPTChannelConfiguration channelConfig = channel.getConfiguration().as(ChatGPTChannelConfiguration.class);
+
+        JsonObject root = new JsonObject();
+        root.addProperty("temperature", channelConfig.temperature);
+        root.addProperty("model", channelConfig.model);
+        root.addProperty("max_tokens", channelConfig.maxTokens);
+
+        JsonObject systemMessage = new JsonObject();
+        systemMessage.addProperty("role", "system");
+        systemMessage.addProperty("content", channelConfig.systemMessage);
+        JsonObject userMessage = new JsonObject();
+        userMessage.addProperty("role", "user");
+        userMessage.addProperty("content", prompt);
+        JsonArray messages = new JsonArray(2);
+        messages.add(systemMessage);
+        messages.add(userMessage);
+        root.add("messages", messages);
+
+        Request request = httpClient.newRequest(OPENAI_API_URL).method(HttpMethod.POST)
+                .header("Content-Type", "application/json").header("Authorization", "Bearer " + apiKey)
+                .content(new StringContentProvider(gson.toJson(root)));
+        try {
+            ContentResponse response = request.send();
+            updateStatus(ThingStatus.ONLINE);
+            if (response.getStatus() == HttpStatus.OK_200) {
+                return response.getContentAsString();
+            } else {
+                logger.error("ChatGPT request resulted in HTTP {} with message: {}", response.getStatus(),
+                        response.getReason());
+                return null;
+            }
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Could not connect to OpenAI API: " + e.getMessage());
+            logger.debug("Request to OpenAI failed: {}", e.getMessage(), e);
+            return null;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        ChatGPTConfiguration config = getConfigAs(ChatGPTConfiguration.class);
+
+        String apiKey = config.apiKey;
+
+        if (apiKey == null || apiKey.isEmpty()) {
```

Comment:
See comment in `ChatGPTConfiguration`.
```suggestion
        if (apiKey.isBlank()) {
```